src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -197,11 +197,7 @@ public IExternalControl Control
 
         public bool HasSelfReference { get; private set; }
 
-        public bool IsBehavior => NameResolver != null && NameResolver.CurrentPropertyIsBehavior;
-
-        public bool IsConstantData => NameResolver != null && NameResolver.CurrentPropertyIsConstantData;
-
-        public bool IsNavigationAllowed => NameResolver != null && NameResolver.CurrentPropertyAllowsNavigation;
```

Comment:
>IsNavigationAllowed [](http://example.com/codeflow?start=20&length=19)

This is used in the Navigate function; are we going to use the same 'AllowSideEffects' flag to deal with it?

Reply:
No, that logic will stay in PA Client. Navigation isn't a concept used in PowerFx, the implementation in Canvas can cast to it's own name resolver impl to retrieve that flag.

Reply:
We have a lot of these "viral" flags... 
where my node has the flag if any of my children have the flag.  Some of these are general and some of them are canvas specific. 
Should we have something generic generic like a bit vector that we just OR together?

- constant, async, navigation,

Reply:
Agreed in general for binder. These (IsConstantData/IsNavigation/IsBehavior) ones are for what the top level property permits and are a separate but similarly named problem.